scout/adapter/mongo/case.py

```diff
@@ -201,94 +193,107 @@ def _set_diagnosis_query(self, query: Dict[str, Any], query_term: str):
+                ),
+                "status": lambda q, v: q.update({"status": v}),
+                "tags": lambda q, v: q.update({"tags": v.lower()}),
+                "track": lambda q, v: q.update({"track": v}),
+                "pheno_group": lambda q, v: q.update({"phenotype_groups.phenotype_id": v}),
+                "cohort": lambda q, v: q.update({"cohorts": v}),
+                "similar_case": lambda q, v: self._set_similar_phenotype_query(
+                    q, "similar_case", v, owner or collaborator
+                ),
+                "similar_pheno": lambda q, v: self._set_similar_phenotype_query(
+                    q, "similar_pheno", v, owner or collaborator
+                ),
+                "pinned": self._set_genes_of_interest_query,
+                "causative": self._set_genes_of_interest_query,
+                "user": set_user_query,
+            }
 
-        if query_term != "" and query_field in ["pinned", "causative"]:
-            self._set_genes_of_interest_query(query, query_field, query_term)
+            handler = handlers.get(query_field)
+            if handler:
+                handler(query, query_value)
 
-        if query_field == "user":
-            query_terms = query_term.split(" ")
+        def set_user_query(query: dict, query_value: str):
+            # Handling the 'user' field query separately
+            query_terms = query_value.split(" ")
             user_query = {
                 "$and": [{"name": {"$regex": term, "$options": "i"}} for term in query_terms]
             }
             users = self.user_collection.find(user_query)
             query["assignees"] = {"$in": [user["_id"] for user in users]}
 
-        return order
+        if isinstance(name_query, str):
+            # Example: status -> Comes from a GET request from dashboard cases
+            query_field, query_value = name_query.split(":", 1)
+            set_case_item_query(
+                query=query, query_field=query_field.strip(), query_value=query_value.strip()
+            )
+        else:
+            # POST request form from more advanced case search from cases page
+            for query_field in [
+                "case",
+                "exact_pheno",
+                "exact_dia",
+                "synopsis",
+                "panel",
+                "status",
+                "tags",
+                "track",
+                "pheno_group",
+                "cohort",
+                "similar_case",
+                "similar_pheno",
+                "pinned",
+                "causative",
+                "user",
+            ]:
+                query_value = name_query.get(query_field)
+                if query_value not in ["", None]:
+                    set_case_item_query(
+                        query=query, query_field=query_field, query_value=query_value.strip()
+                    )
 
-    def _update_case_id_query(self, query, id_list):
+    def update_case_id_query(self, query, id_list):
```

Comment:
Huh, is this really supposed to be imported outside the module now? Or am I being archaic with PEP-8?
![Screenshot 2024-08-28 at 08 12 57](https://github.com/user-attachments/assets/49287e0d-960b-48b1-8316-fcb18d799606)

Reply:
Mmm no, perhaps I intended to import it from some other place at one point, but I didn't do it. I'll revert!